nodebuilder/das/module.go

```diff
@@ -11,11 +11,31 @@ import (
 	"github.com/celestiaorg/celestia-node/nodebuilder/node"
 )
 
-func ConstructModule(tp node.Type) fx.Option {
+func ConstructModule(tp node.Type, cfg *Config) fx.Option {
+
+	cfgErr := cfg.Validate()
+
+	baseComponents := fx.Options(
+		fx.Supply(*cfg),
+		fx.Error(cfgErr),
+		fx.Provide(
+			func(c Config) []das.Option {
+				return []das.Option{
+					das.WithSamplingRange(c.SamplingRange),
+					das.WithConcurrencyLimit(c.ConcurrencyLimit),
+					das.WithPriorityQueueSize(c.PriorityQueueSize),
+					das.WithBackgroundStoreInterval(c.BackgroundStoreInterval),
+					das.WithGenesisHeight(c.GenesisHeight),
+				}
+			},
+		),
+	)
+
 	switch tp {
 	case node.Light, node.Full:
 		return fx.Module(
 			"daser",
+			baseComponents,
 			fx.Provide(fx.Annotate(
 				NewDASer,
```

Comment:
@renaynay I want to double-check, we changed the signature for this constructor to return an error, if this errors out, will fx halt the instantion and `app.Err()` should return the error causing the halt, is this correct?

Reply:
I am not Rene, but def yes